libsnark/gadgetlib1/gadgets/hashes/anemoi/tests/test_anemoi_gadget.cpp

```diff
@@ -0,0 +1,221 @@
+/** @file
+ *****************************************************************************
+ * @author     This file is part of libsnark, developed by SCIPR Lab
+ *             and contributors (see AUTHORS).
+ * @copyright  MIT license (see LICENSE file)
+ *****************************************************************************/
+
+#include <gtest/gtest.h>
+#include <libff/algebra/curves/bls12_381/bls12_381_init.hpp>
+#include <libff/algebra/curves/bls12_381/bls12_381_pp.hpp>
+#include <libff/common/default_types/ec_pp.hpp>
+#include <libff/common/profiling.hpp>
+#include <libff/common/utils.hpp>
+#include <libsnark/common/default_types/r1cs_ppzksnark_pp.hpp>
+#include <libsnark/gadgetlib1/gadgets/hashes/anemoi/anemoi_components.hpp>
+#include <libsnark/gadgetlib1/gadgets/hashes/anemoi/anemoi_constants.hpp>
+#include <libsnark/zk_proof_systems/ppzksnark/r1cs_gg_ppzksnark/r1cs_gg_ppzksnark.hpp>
+//#include
+//<libsnark/zk_proof_systems/ppzksnark/r1cs_ppzksnark/r1cs_ppzksnark.hpp>
+
+using namespace libsnark;
+
+template<typename ppT>
+void test_pb_verify_circuit(protoboard<libff::Fr<ppT>> &pb)
+{
+    ASSERT_TRUE(pb.is_satisfied());
+    const r1cs_gg_ppzksnark_keypair<ppT> keypair =
+        r1cs_gg_ppzksnark_generator<ppT>(pb.get_constraint_system(), true);
+    r1cs_primary_input<libff::Fr<ppT>> primary_input = pb.primary_input();
+    r1cs_auxiliary_input<libff::Fr<ppT>> auxiliary_input = pb.auxiliary_input();
+    r1cs_gg_ppzksnark_proof<ppT> proof = r1cs_gg_ppzksnark_prover(
+        keypair.pk, primary_input, auxiliary_input, true);
+    ASSERT_TRUE(r1cs_gg_ppzksnark_verifier_strong_IC<ppT>(
+        keypair.vk, primary_input, proof));
+}
+
+template<typename ppT>
+void test_flystel_Q_gamma_prime_field_gadget(const size_t n)
+{
+    using FieldT = libff::Fr<ppT>;
+    printf("testing flystel_power_two_gadget on all %zu bit strings\n", n);
```

Comment:
Can be useful for development, but the testing framework shoudl tell us which tests has failed, and how.

Reply:
I added this info to be consistent with the format in other tests e.g. https://github.com/clearmatics/libsnark/blob/develop/libsnark/gadgetlib1/tests/test_basic_gadgets.cpp#L19 . Should I remove it?

Reply:
If you no longer need it I think it's best to remove it.  It clutters up the output when running the test. (At least when running manually.  I think gtest hides the stdout by default.)

Reply:
Fixed.